src/common/mfu_flist_copy.c

```diff
@@ -1981,8 +1999,150 @@ static int mfu_copy_file_fiemap(
         }
     }
 
+    for (ssize_t idx = 0; idx < fiemap->fm_mapped_extents; idx++) {
+        extent_list->mel_extents[idx].me_logical = fiemap->fm_extents[idx].fe_logical;
+        extent_list->mel_extents[idx].me_length = fiemap->fm_extents[idx].fe_length;
+    }
+    extent_list->mel_mapped_extents = fiemap->fm_mapped_extents;
+
+    free(fiemap);
+
     *normal_copy_required = false;
 
+    return extent_list;
+
+fail_free_fiemap:
+    free(fiemap);
+    free(extent_list);
+
+fail_no_fiemap:
+    return NULL;
+}
+
+static struct mfu_extent_list * mfu_lseek_get_extents(
+    const char* src,
+    const char* dest,
+    uint64_t offset,
+    uint64_t length,
+    uint64_t file_size,
+    bool* normal_copy_required,
+    mfu_copy_opts_t* copy_opts,
+    mfu_file_t* mfu_src_file,
+    mfu_file_t* mfu_dst_file)
+{
+    *normal_copy_required = true;
+    uint64_t last_byte = offset + length;
+    int op;
+    int idx;
+    uint64_t start;
+
+    /* initial guess at max number of extents, can revise */
+    size_t extent_count = 16;
+    struct mfu_extent_list* extent_list = mfu_extent_list_realloc(NULL, extent_count);
+    if (extent_list == NULL) {
+        MFU_LOG(MFU_LOG_ERR, "Out of memory allocating extent_list with count %d", extent_count);
+        goto fail_no_extent_list;
+    }
+
+    op = SEEK_DATA;
+    idx = 0;
+
+    MFU_LOG(MFU_LOG_DBG, "map from lseek():");
+    start = offset;
+    while (offset < last_byte) {
+        offset = lseek(mfu_src_file->fd, offset, op);
+
+        if (op == SEEK_HOLE) {
+            MFU_LOG(MFU_LOG_DBG, "%llu - %llu type %s\n", start, offset, (op == SEEK_HOLE ? "data" : "hole"));
```

Comment:
the ternary operator condition is always true here, "op == SEEK_HOLE" is always true inside this if block.

Reply:
I'll just remove that message.